ste/sender-blockBlob.go

```diff
@@ -153,19 +153,38 @@ func (s *blockBlobSenderBase) Epilogue() {
 	// https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers
 	if jptm.IsLive() && s.destBlobTier != azblob.AccessTierNone {
 		// Set the latest service version from sdk as service version in the context.
-		ctxWithLatestServiceVersion := context.WithValue(jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
-		_, err := s.destBlockBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{})
-		if err != nil {
-			if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
-				s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-				s2sAccessTierFailureLogStdout.Do(func() {
-					glcm := common.GetLifecycleMgr()
-					glcm.Error("One or more blobs have failed blob tier replication at the destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-				})
+		ctxWithLatestServiceVersion := context.WithValue(s.jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
+
+		// Let's check if we can confirm we'll be able to check the destination blob's account info.
+		// A SAS token, even with write-only permissions is enough. OR, OAuth with the account owner.
+		// We can't guess that last information, so we'll take a gamble and try to get account info anyway.
+		destParts := azblob.NewBlobURLParts(s.destBlockBlobURL.URL())
+		mustGet := destParts.SAS.Encode() != ""
+
+		prepareDestAccountInfo(s.destBlockBlobURL.BlobURL, s.jptm, ctxWithLatestServiceVersion, mustGet)
+		tierAvailable := BlobTierAllowed(s.destBlobTier)
+
+		if tierAvailable {
+			_, err := s.destBlockBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{})
+			if err != nil {
+				if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
+					s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
```

Comment:
Does having a SrcAccess tier really proved that the thing that failed was the attempt to replicate (i.e. copy) the tier?  
 I don't think it does prove that, because what if the user had set an explicit destination tier using a command line parameter? They could have specified one that's not available at the destination.  But error messages saying "replicate" would be confusing in that case, because replicate means "copy", but this is not a case of copying the tier from the source.

If we want to give an error specific to replication, I would have expected the if statement to say something like `if s.jptm.Info().S2SSrcBlobTier == s.destBlobTier`.  That will return true if we are indeed _replicating_ the tier.

(When writing the above, I'm _assuming_ that the command line parameters affect s.destBlobTier but not S2SSRcBlobTier. You should check my assumption before relying on it ;-)

Reply:
Your assumption was right, and I've implemented it.